### PR TITLE
Enhancement: Tweak logging for artist facing reports

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Tweak the logging of publishing for global, deadline, maya and a fusion plugin to have a cleaner artist-facing report.
- Fix context being reported correctly from CollectContext
- Fix ValidateMeshArnoldAttributes: fix when arnold is not loaded, fix applying settings, fix for when ai attributes do not exist

## Additional info

This touches so many logs that it's probably best to do a code review - except for the areas that feel like changes that are more than logging level changes or cosmetics (there are some tiny bugfixes or tweaks on how things are logged as I cleaned it up.)

I can't seem to assign a _Chore_ label so will mark this _Enhancement_ since it improves artist experience with new publisher.
 
## Testing notes:

1. Publish some instances across a myriad of hosts, especially maya and deadline publishing.

### CODE REVIEW

**Review by LiborBatek — COMMENTED** [review]
How about this one? Speaking of publishing ```model``` and Arnold renderer not being loaded at the moment :)
image

**Comment by LiborBatek** [comment]
...also speaking of artists pov and readibility of the logs....I m always confused by list of invalid objects in this way (full hierarchy listed but actual objects being somewhat hidden "between the lines" ...if any way to separate those somehow

image


maybe just the geometry names as in Outliner (in separate list or so)

Just a thought...ignore if not applicable :)

**Comment by LiborBatek** [comment]
Haha going deeper with Publisher logs...I can see more and more things to be tweaked....basically Im not enjoying it much as things are being now regarding the way it shows various errors/issues.

There is huge room for UX improvement there...but I guess its a different story for different PR :)

**Comment by LiborBatek** [comment]
@BigRoy is it possible that those updates are valid only for publish instances created with newer OP versions (when new Publisher was introduced) ??

Because Im experiencing some of my scenes having pretty nice Reports aka Logs like this one which I do like a lot!

image

**Comment by BigRoy** [comment]
Actually with https://github.com/ynput/OpenPype/pull/5537/commits/689da974c4d22161797862f2a9e77d8aeee65656 I've now also fixed that `ValidateMeshArnoldAttributes` actually adheres to the Plug-in's settings in Project Settings instead of being active or not just based on whether the model workfile just happened to have "arnold" as current renderer or not.

Note that this now means that `ValidateMeshArnoldAttributes` is disabled by default (it doesn't run) because the Default Project Settings of OpenPype have the validator disabled. See [the default value in settings here](https://github.com/ynput/OpenPype/blob/dd27f4e839abe491d54b0183b58dc0dbbe16dc6a/openpype/settings/defaults/project_settings/maya.json#L966).

**Comment by iLLiCiTiT** [comment]
If there would be only changes to `debug` I would approve and merge it, but with the other changes it has to wait for @antirotor, who can review the content changes.

**Review by Minkiu — APPROVED** [review]
Haven't tested it, but code looks good.

File: openpype/plugins/publish/extract_review_slate.py [hunk 1]
```diff
@@ -14,10 +14,13 @@
     get_ffmpeg_codec_args,
     get_ffmpeg_format_args,
 )
-from openpype.pipeline import publish
+from openpype.pipeline.publish import (
+    Extractor,
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/extract_review_slate.py:18]: This plugin will be discovered and processed.

Comment (BigRoy) [reply openpype/plugins/publish/extract_review_slate.py]: Fixed with https://github.com/ynput/OpenPype/pull/5537/commits/564cb31cbe700aef687f6b11969eb530973e60d9

Comment (BigRoy) [reply openpype/plugins/publish/extract_review_slate.py]: By the way @iLLiCiTiT - this particular bug actually **hangs** the publisher. The publishing just stops. It doesn't go to an error page, it doesn't go to a report page - only in the Maya script editor I see "raise ValueError". The user can stop the publisher (it doesn't hang the actual UI; it just stops doing anything remaining visually in the state as if it's still doing something).
```python
# Traceback (most recent call last):
#   File "C:\Users\Maqina-RTX-05\AppData\Local\pypeclub\openpype\3.16\openpype-v3.16.5-colorbleed.2\openpype\tools\publisher\control_qt.py", line 56, in _execute
#     item.process()
#   File "C:\Users\Maqina-RTX-05\AppData\Local\pypeclub\openpype\3.16\openpype-v3.16.5-colorbleed.2\openpype\tools\publisher\control.py", line 64, in process
#     self.callback(*self.args, **self.kwargs)
#   File "C:\Users\Maqina-RTX-05\AppData\Local\pypeclub\openpype\3.16\openpype-v3.16.5-colorbleed.2\openpype\tools\publisher\control.py", line 2519, in _process_and_continue
#     self._publish_next_process()
#   File "C:\Users\Maqina-RTX-05\AppData\Local\pypeclub\openpype\3.16\openpype-v3.16.5-colorbleed.2\openpype\tools\publisher\control.py", line 2340, in _publish_next_process
#     item = next(self._main_thread_iter)
#   File "C:\Users\Maqina-RTX-05\AppData\Local\pypeclub\openpype\3.16\openpype-v3.16.5-colorbleed.2\openpype\tools\publisher\control.py", line 2411, in _publish_iterator
#     self._publish_report.add_plugin_iter(
#   File "C:\Users\Maqina-RTX-05\AppData\Local\pypeclub\openpype\3.16\openpype-v3.16.5-colorbleed.2\openpype\tools\publisher\control.py", line 213, in add_plugin_iter
#     self._current_plugin_data = self._add_plugin_data_item(plugin)
#   File "C:\Users\Maqina-RTX-05\AppData\Local\pypeclub\openpype\3.16\openpype-v3.16.5-colorbleed.2\openpype\tools\publisher\control.py", line 229, in _add_plugin_data_item
#     raise ValueError(
# ValueError: Plugin '<class 'C:\Users\Maqina-RTX-05\AppData\Local\pypeclub\openpype\3.16\openpype-v3.16.5-colorbleed.2\openpype\plugins\publish\extract_review_slate.py.Extractor'>' is already stored
```

Seems like at least that's a case the Publisher should handle prettier. I'd expect at least the "Report your IT" message.

**Review by antirotor — APPROVED** [review]
Loogs good.

**Pull request merged**